vulnfeeds/cves/nvd2.go

```diff
@@ -0,0 +1,418 @@
+		return fmt.Errorf("field %s length: must be <= %d", "url", 500)
+	}
+	*j = Reference(plain)
+	return nil
+}
+
+type VendorComment struct {
+	// Comment corresponds to the JSON schema field "comment".
+	Comment string `json:"comment" yaml:"comment" mapstructure:"comment"`
+
+	// LastModified corresponds to the JSON schema field "lastModified".
+	LastModified NVDTime `json:"lastModified" yaml:"lastModified" mapstructure:"lastModified"`
+
+	// Organization corresponds to the JSON schema field "organization".
+	Organization string `json:"organization" yaml:"organization" mapstructure:"organization"`
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (j *LangString) UnmarshalJSON(b []byte) error {
+	var raw map[string]interface{}
+	if err := json.Unmarshal(b, &raw); err != nil {
+		return err
+	}
+	if v, ok := raw["lang"]; !ok || v == nil {
+		return fmt.Errorf("field lang in LangString: required")
+	}
+	if v, ok := raw["value"]; !ok || v == nil {
+		return fmt.Errorf("field value in LangString: required")
+	}
+	type Plain LangString
+	var plain Plain
+	if err := json.Unmarshal(b, &plain); err != nil {
+		return err
+	}
+	// CVE-2020-35800 runs afoul of this (see https://github.com/omissis/go-jsonschema/issues/155#issuecomment-1823970902)
+	// if len(plain.Value) > 4096 {
+	// 	return fmt.Errorf("field %s length: must be <= %d, have %d with content %q", "value", 4096, len(plain.Value), plain.Value)
+	// }
+	*j = LangString(plain)
+	return nil
+}
+
+type Weakness interface{}
+
+type LangString struct {
+	// Lang corresponds to the JSON schema field "lang".
+	Lang string `json:"lang" yaml:"lang" mapstructure:"lang"`
+
+	// Value corresponds to the JSON schema field "value".
+	Value string `json:"value" yaml:"value" mapstructure:"value"`
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (j *CVEItem) UnmarshalJSON(b []byte) error {
+	var raw map[string]interface{}
+	if err := json.Unmarshal(b, &raw); err != nil {
+		return err
+	}
+	if v, ok := raw["descriptions"]; !ok || v == nil {
+		return fmt.Errorf("field descriptions in CveItem: required")
+	}
+	if v, ok := raw["id"]; !ok || v == nil {
+		return fmt.Errorf("field id in CveItem: required")
+	}
+	if v, ok := raw["lastModified"]; !ok || v == nil {
+		return fmt.Errorf("field lastModified in CveItem: required")
+	}
+	if v, ok := raw["published"]; !ok || v == nil {
+		return fmt.Errorf("field published in CveItem: required")
+	}
+	if v, ok := raw["references"]; !ok || v == nil {
+		return fmt.Errorf("field references in CveItem: required")
+	}
+	type Plain CVEItem
+	var plain Plain
+	if err := json.Unmarshal(b, &plain); err != nil {
+		return err
+	}
+	if plain.Descriptions != nil && len(plain.Descriptions) < 1 {
+		return fmt.Errorf("field %s length: must be >= %d", "descriptions", 1)
+	}
+	if len(plain.References) > 500 {
+		return fmt.Errorf("field %s length: must be <= %d", "references", 500)
+	}
+	*j = CVEItem(plain)
+	return nil
+}
+
+// (hand generated).
+type DefCveItem struct {
```

Comment:
nit: s/Cve/CVE/.

Also, can we have a more descriptive comment? It's also not clear what "Def" is short for here.

Reply:
If there's no interest in retaining (re)generatability, we can rename away from the generated names. I agree this one's unfortunately named due to the way the underlying schema is expressed.

Reply:
Renamed for better readability in https://github.com/google/osv.dev/pull/1833/commits/e5bed10b2bd6973ad2a063fffc38780533f8cd3b